src/sign_dlink_ru.c

```diff
@@ -91,6 +90,11 @@ int asciihex_to_int(char c) {
 void verify_valid_hex_str(char* s) {
 	int i;
 	int s_len = strlen(s);
+	
+	if (strncmp(s,"-",1) == 0) { // Don't check -switch
+		s = "00" ; // dummy to pass
```

Comment:
instead of setting s to something that is valid hex, just `return` and skip the checks.

Reply:
sounds good just don't know the language enough
i have more hacked it then programmed it